plugins/home/src/homePageComponents/VisitedByType/Content.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { VisitedByType } from './VisitedByType';
+import { Visit, visitsApiRef } from '../../api/VisitsApi';
+import { ContextValueOnly, useContext } from './Context';
+import { useApi } from '@backstage/core-plugin-api';
+import useAsync from 'react-use/lib/useAsync';
+
+/** @public */
+export type VisitedByTypeKind = 'recent' | 'top';
+
+/** @public */
+export type VisitedByTypeProps = {
+  visits?: Array<Visit>;
+  numVisitsOpen?: number;
+  numVisitsTotal?: number;
+  loading?: boolean;
+  kind: VisitedByTypeKind;
+};
+
+/**
+ * Display recently visited pages for the homepage
+ * @public
+ */
+export const Content = ({
+  visits,
+  numVisitsOpen,
+  numVisitsTotal,
+  loading,
+  kind,
+}: VisitedByTypeProps) => {
+  const { setContext, setVisits, setLoading } = useContext();
+  // Allows behavior override from properties
+  useEffect(() => {
+    const context: Partial<ContextValueOnly> = {};
+    context.kind = kind;
+    if (visits) {
+      context.visits = visits;
+      context.loading = false;
+    } else if (loading) {
+      context.loading = loading;
+    }
+    if (numVisitsOpen) context.numVisitsOpen = numVisitsOpen;
+    if (numVisitsTotal) context.numVisitsTotal = numVisitsTotal;
+    setContext(state => ({ ...state, ...context }));
+  }, [setContext, kind, visits, loading, numVisitsOpen, numVisitsTotal]);
+
+  // Fetches data from visitsApi in case visits and loading are not provided
+  const visitsApi = useApi(visitsApiRef);
+  const { loading: reqLoading } = useAsync(async () => {
+    if (!visits && !loading && kind === 'recent') {
+      return await visitsApi
+        .listUserVisits({
+          limit: numVisitsTotal ?? 8,
+          orderBy: { timestamp: 'desc' },
+        })
+        .then(setVisits);
+    }
+    if (!visits && !loading && kind === 'top') {
+      return await visitsApi
+        .listUserVisits({
+          limit: numVisitsTotal ?? 8,
+          orderBy: { hits: 'desc' },
+        })
+        .then(setVisits);
+    }
+    return undefined;
+  }, [visitsApi, visits, loading, setVisits]);
+  useEffect(() => {
+    if (!loading) {
+      setLoading(reqLoading);
+    }
+  }, [loading, setLoading, reqLoading]);
+
+  return <VisitedByType />;
```

Comment:
I don't see much point in creating a context here just to separate the api call bit from the view. You could just render VisitList here instead of creating a separate VisitedByType component in a separate file where the context gets used. There aren't other components that benefit from this context any way. It makes the code more complex than simple imo.

Reply:
The button declared on `Action` needs to open and close the visible items. It also requires to know values of loading, visits.length and number of open visits for another logic (hide the button if necessary).
The context was created to sort ou this integration. The other parameters are not required, true, my decision here was to keep eveything in the context to handle all parameters the same.
If that is aggregious to you, I will be happy to accept a PR to change it.

Reply:
I might be blind, I can't seem to find the place where Actions is used in the visited list implementation :/

Reply:
https://github.com/backstage/backstage/pull/19645/files#diff-0335c3a7e723b8fd1d8e546505b3dc0af9dcf510eaadabc6660ee5f68fb0ec10